stories/Button/Button.tsx

```diff
@@ -1,31 +1,123 @@
-import React from "react";
+import { MouseEvent } from "react";
 
 interface ButtonProps {
   /**
-   * What background color to use
+   * Estilos de botón.
    */
-  backgroundColor?: string;
+  variant?: "primary" | "secondary" | "outlined" | "icon";
+  /**
+   * Tamaño del botón
+   */
+  size?: "small" | "medium" | "large" | "stretched";
+  /**
+   * Label requerida para describir el botón.
+   */
+  ariaLabel: string;
+  /**
+   * Label opcional para describiendo el botón.
+   */
+  label?: string;
+  /**
+   * Indica que el botón esta deshabilitad, esto previene que se pueda hacer click en el botón.
+   */
+  disabled?: boolean;
+  /**
+   * Indica que el botón esta cargando, esto previene que se pueda hacer click en el botón.
+   */
+  loading?: boolean;
+  /**
+   * Tipo de botón, en este caso por defecto es "button".
+   */
+  type: "submit" | "reset" | "button";
+  /**
+   * El icono para mostrar en el botón. Por defecto se muestra a la izquierda del texto (si este existe).
+   */
+  icon?: React.ReactNode;
   /**
-   * Button contents
+   * El icono para mostrar en el botón. Por defecto se muestra a la derecha del texto (si este existe).
    */
-  label: string;
+  rightIcon?: React.ReactNode;
   /**
-   * Optional click handler
+   * Borde redondeado del botón. Por defecto es "md".
    */
-  onClick?: () => void;
+  rounded?: "base" | "md" | "lg";
+  /**
+   * Click handler requerido para el botón
+   */
+  onClickHandler: (event: MouseEvent<HTMLButtonElement>) => void;
+
+  children?: React.ReactNode;
+
+  // TODO: These two are only for the storybook. Remove it when we have a proper styles for it.
+  /**
+   * Color de fondo de botón
+   */
+  backgroundColor?: string;
```

Comment:
Quiza el fondo del boton se puede definir desde el variant. Ej. Si es primary, entonces el fondo es blanco, si es secondary, entonces es verde, etc. De esa forma podriamos eliminar esta prop.

Reply:
Ah si, esto lo deje solo porque me parecio interesante el feature de Storybook de elegir colores (de hecho deje un "to-do" para eliminarlo junto con `color`, enseguida los saco)

![Screenshot_20220812_220343](https://user-images.githubusercontent.com/78808163/184464545-53374ba1-932a-4cce-ac82-1bbadbdef1c6.png)